torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -286,9 +476,19 @@ def affine_image_tensor(
         return image
 
     shape = image.shape
-    num_channels, height, width = shape[-3:]
-    image = image.reshape(-1, num_channels, height, width)
+    ndim = image.ndim
+    fp = torch.is_floating_point(image)
 
+    if ndim > 4:
+        image = image.reshape((-1,) + shape[-3:])
+        needs_unsquash = True
+    elif ndim == 3:
+        image = image.unsqueeze(0)
+        needs_unsquash = True
+    else:
+        needs_unsquash = False
```

Comment:
Adopting the idiom from other places to handle squashing. I do the same in all methods where possible below.

Reply:
Do we need the special casing for `ndim == 3` here? Is there actually a significant perf difference between `reshape` and `unsqueeze` in this case?

Reply:
This is not a perf. This is because `_apply_grid_transform` assumes batched input.

Reply:
As discussed offline, this is copied from something so not a new idiom that will be introduced here. I'll look into it in a follow-up.